File: services/bff/src/presentation/presenters/agent.presenter.ts
import { AgentOutput } from '../../application/dtos';
import { AgentDto } from '../graphql/dto/agent.dto';

export class AgentPresenter {
  static toGraphql(agent: AgentOutput): AgentDto {
    return {
      id: agent.id,
      name: agent.name,
    };
  }
}

File: services/agent/src/presentation/presenters/agent_presenter.py
from src.application.dtos.agent_dto import AgentOutput
from src.presentation.grpc.generated import agents_pb2


class AgentPresenter:
    @staticmethod
    def to_proto(output: AgentOutput) -> agents_pb2.Agent:
        return agents_pb2.Agent(
            id=output.id,
            conversation_id=output.conversation_id,
            name=output.name,
            system_prompt=output.system_prompt,
            model=output.model,
            created_at=output.created_at,
            updated_at=output.updated_at,
        )
